### ISSUE TITLE
macOS Catalina: `system_info` table is consistently slow

### ISSUE THREAD

**Issue by sharvilshah** [osquery/osquery#5873]
# Bug report

Queries to `system_info` table are consistently taking 5+ seconds on macOS 10.15 Catalina. 

```
$ ./osqueryd -S --line "select * from system_info;" 
 0.04s user 0.01s system 0% cpu 6.368 total

$ ./osqueryd -S --line "select * from system_info;"  
 0.04s user 0.01s system 0% cpu 5.062 total

osquery> select * from system_info;
...
Run Time: real 5.008 user 0.001493 sys 0.001089
```

I noticed this both on VMs with 10.15 and on real hardware.
On machines running 10.13 and 10.14 the same query takes less than half a second.

I tried tracing this, but nothing seems to jump out. And the implementation of `system_info` also hasn't changed. My hunch is that maybe something with the system APIs we use changed in 10.15, and if that's the case there maybe not be much we could do about it -- I will try to dig more later.

### What operating system and version are you using?

```
 version = 10.15
   build = 19A583
platform = darwin
```

### What version of osquery are you using?

3.3.2, 4.0.2 and current `master`

### What steps did you take to reproduce the issue?

Run `select * from system_info` (or any column(s) from `system_info`) on 10.15

### What did you expect to see?

The query taking little under a second to run.

### What did you see instead?

The query taking more than 5 seconds to run.

**Comment by sharvilshah** [comment]
This currently seems endemic to my machine. I am keeping this open for a little longer till I have a chance to attach a debugger and take a look.

**Comment by directionless** [comment]
On my catalina machine, I ran `time osqueryd -S  'select * from system_info'`. It took  0.04s. I'll try via the interactive shell tomorrow

**Comment by directionless** [comment]
Yes, still returns promptly on my catalina machine. Interactive shell or straight command line. This is build `19A582a`

**Comment by mike-myers-tob** [comment]
Using 4.5.0 and Seph's command, I get 0.04s too, on macOS 10.15.6.

I nominate closing this issue, and it can be reopened if @sharvilshah can give more conditions to reproduce?

**Comment by sharvilshah** [comment]
Yes, closing this, this is no longer an issue :) 
IIRC this resolved itself with 10.15.2
